### DISCUSSION TITLE
Github actions workflow trigger from JIRA board/Ticket

### DISCUSSION THREAD

**Discussion by Srikanth419** [community/community#63704] [Actions]
### Select Topic Area

Show & Tell

### Body

I have GitHub workflow pipeline. I want to trigger/execute that pipeline from Jira board.
If that possible to execute pipeline for JIRA board

**Comment by nirav-gajera** [comment]
Yes, it is possible to trigger/execute a GitHub workflow pipeline from a Jira board. To do this, you will need to use the GitHub Action called `workflow_dispatch`. This action allows you to trigger a workflow pipeline from any event, including a Jira issue being created or updated.


```
on:
  issue_created:
    types: [opened]

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/workflow_dispatch@v2
        with:
          workflow_file: .github/workflows/my-pipeline.yaml
```

In this example, the `workflow_dispatch` action is triggered when a Jira issue is created. The `workflow_file` input is set to the path to the GitHub workflow file that you want to trigger.

**Reply by Srikanth419** [reply]
Hello nirav
Thanks for your response
Can you explain how the link between Github and Jira established.
How the triggering happening from Jira board

**Reply by nirav-gajera** [reply]
Sure., create a connection between your GitHub repository and your Jira project. Once the connection is established, you can trigger GitHub workflows from Jira issues.

To trigger a GitHub workflow from a Jira issue, you will need to add a comment to the issue with the following syntax:

```
/github-workflow <workflow_name>
```

For example, to trigger the workflow called `my-pipeline` from a Jira issue, you would add the following comment to the issue:

```
/github-workflow my-pipeline
```

The `workflow_name` is the name of the GitHub workflow file that you want to trigger. The workflow file must be located in the `.github/workflows` directory of your GitHub repository.

Once you have added the comment to the Jira issue, the GitHub workflow will be triggered. The workflow will run the steps that are defined in the workflow file.

Here are the steps on how to establish the link between GitHub and Jira:

1. Go to the GitHub App Marketplace and search for "Jira Integration".
2. Click on the "Install" button.
3. Follow the instructions to authorize the app to access your GitHub repository and Jira project.
4. Once the app is authorized, you will see a new "Jira" tab in your GitHub repository.
5. In the "Jira" tab, you can see a list of your Jira projects.
6. Click on the name of the Jira project that you want to connect to.
7. In the "Settings" section, you can configure the connection between GitHub and Jira.
8. Once you have configured the connection, you can start triggering GitHub workflows from Jira issues.

**Comment by Srikanth419** [comment]
Hello Nirav
https://github.com/marketplace/jira-integration-for-github
Can you confirm is this https://github.com/marketplace/jira-integration-for-github same you mentioned?

**Reply by nirav-gajera** [reply]
yes go for it

**Comment by Srikanth419** [comment]
I am using JIRA Server not Jira Cloud will it work for Jira Server.

**Comment by Jaber-Saed** [comment]
Yes, it is possible to trigger a GitHub Actions workflow pipeline from a Jira board. However, this requires a bit of setup involving webhooks and automation.

Here's a general overview of how you can achieve this:

1. **Configure Webhook in Jira:**
   - In your Jira board, navigate to the "Settings" or "Admin" section.
   - Look for the "Automation" or "Workflows" option (the exact name may vary depending on your Jira version).
   - Create a new automation rule or workflow that will trigger when a certain condition is met, such as when a new issue is created or when an issue is moved to a certain status.

2. **Create Webhook in GitHub:**
   - In your GitHub repository, navigate to "Settings" > "Webhooks" > "Add webhook."
   - Set the payload URL to the GitHub Actions webhook URL for your repository. This URL should look like this: `https://api.github.com/repos/:owner/:repo/dispatches`
   - Set the content type to `application/json`.
   - Generate a secret token and provide it in the webhook settings. This secret token will be used to verify requests from Jira.

3. **Automation Rule:**
   - Configure the automation rule in Jira to trigger the webhook when a certain event happens (e.g., issue creation, status change).
   - When configuring the webhook, pass the required payload data in JSON format. This payload will contain information about the issue or event that triggered the webhook.

4. **GitHub Actions Workflow:**
   - In your GitHub repository, create a GitHub Actions workflow YAML file (e.g., `.github/workflows/jira-trigger.yml`).
   - Configure a custom event trigger using the `workflow_dispatch` event.
   - Define the steps you want to run in response to the trigger.

Here's an example of how the workflow YAML file might look:

```yaml
name: Jira Trigger

on:
  repository_dispatch:
    types: [jira]

jobs:
  jira-trigger:
    runs-on: ubuntu-latest

    steps:
      - name: Check out code
        uses: actions/checkout@v2

      # Add your workflow steps here
```

In this example, the `repository_dispatch` event is used to trigger the workflow when a Jira webhook sends a payload with the `jira` type. You'll need to replace `jira` with the type you specify in your Jira webhook configuration.

Remember that this is a high-level overview, and the exact steps might vary based on your Jira and GitHub configurations. It's important to review the documentation for both Jira and GitHub Actions to ensure you set up the webhook and workflow correctly.

Additionally, keep security in mind when setting up webhooks and sharing secrets. Make sure to follow best practices for securing your integrations.

**Comment by Srikanth419** [comment]
hello @Jaber-Saed 

Thanks for your help
I am able to trigger github actions pipeline form Jira ticket when ever status change in the JIRA ticket using repository_dispatch event
Noe I want to pass Jira field values as variables in GitHub Actions pipeline 
I there any way that I can pass Jira field values to pipeline.

**Reply by ccd-bamparo** [reply]
Did you ever find a workaround for this? I want to do the exact same thing.

**Reply by evalue-simon-harrison** [reply]
> Did you ever find a workaround for this? I want to do the exact same thing.

Same - I'd love to know if it's possible
